test/app-luatest/override_misc_test.lua

```diff
@@ -0,0 +1,54 @@
+local t = require('luatest')
+local treegen = require('test.treegen')
+local justrun = require('test.justrun')
+
+local g = t.group()
+
+local OVERRIDE_SCRIPT_TEMPLATE = [[
+local loaders = require('internal.loaders')
+
+return {
+    whoami = 'override.<module_name>',
+    initializing = loaders.initializing,
+}
+]]
+
+-- Print a result of the require call.
+local MAIN_SCRIPT_TEMPLATE = [[
+local json = require('json')
+
+print(json.encode(require('<module_name>')))
+]]
+
+g.before_all(function(g)
+    treegen.init(g)
+    treegen.add_template(g, '^override/.*%.lua$', OVERRIDE_SCRIPT_TEMPLATE)
+    treegen.add_template(g, '^main%.lua$', MAIN_SCRIPT_TEMPLATE)
+end)
+
+g.after_all(function(g)
+    treegen.clean(g)
+end)
+
+local function expected_output(module_name)
+    local res = {
+        {
+            whoami = ('override.%s'):format(module_name),
+            initializing = true,
+        }
+    }
+
+    return {
+        exit_code = 0,
+        stdout = res,
+    }
+end
+
```

Comment:
I suggest also to add some base test, that `initializing` is not true, when we run this code.

Reply:
Kinda this?

```diff
diff --git a/test/app-luatest/override_misc_test.lua b/test/app-luatest/override_misc_test.lua
index 47cf317a2..0287810b0 100644
--- a/test/app-luatest/override_misc_test.lua
+++ b/test/app-luatest/override_misc_test.lua
@@ -16,8 +16,15 @@ return {
 -- Print a result of the require call.
 local MAIN_SCRIPT_TEMPLATE = [[
 local json = require('json')
+local loaders = require('internal.loaders')
 
-print(json.encode(require('<module_name>')))
+print(json.encode({
+    ['<module_name>'] = require('<module_name>'),
+    ['<script>'] = {
+        whoami = '<script>',
+        initializing = loaders.initializing,
+    }
+}))
 ]]
 
 g.before_all(function(g)
@@ -33,8 +40,14 @@ end)
 local function expected_output(module_name)
     local res = {
         {
-            whoami = ('override.%s'):format(module_name),
-            initializing = true,
+            [module_name] = {
+                whoami = ('override.%s'):format(module_name),
+                initializing = true,
+            },
+            ['main.lua'] = {
+                whoami = 'main.lua',
+                -- initializing is nil
+            }
         }
     }
```

Added.

Reply:
Yep, it's OK.